### PR TITLE
Nathan/file preview

### PR DESCRIPTION
**Pull request by nathandtam**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Implemented `/api/sqlite/files/[id]` API route to serve locally downloaded files from disk.
- Updated sync route to download files from Supabase bucket and save them to `/home/<name-of-pi>/rose-files`.
- Use Node streaming and mime type detection to download files.
- Created a minimal FilePreview component that embeds the file in an `<iframe>`.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2025-11-04 at 2 47 21 AM

## 👍 How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

To test the sync and file downloads:
1. Inside the project root, do mkdir rose-files.
2. Add the line `LOCAL_FILES_DIR=./rose-files` to your `.env.local`.
3. Run `pnpm dev` and go to http://localhost:3000/api/sync.
4. The files should be downloaded to your rose-files folder within your project.

To view the files on the app (note: this must be done after sync, so that you have the files locally):
1. Keep serving running (run `pnpm dev` if it isn't).
2. Make sure that you have synced.
3. Go to http://localhost:3000/api/sqlite/files/24 where the '24' is the ID of one of the files.

If you want to view the file preview, create a test page and use the `FilePreview` component in it.

## 👍 Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Style the component once the other files page is more complete.
- When creating the local tables, we use ALTER to add the new columns we need in order to accommodate for two cases: Pis that already have the tables created (other developers have worked on them already) and Pis that don't have the tables created yet (new and blank). This is not necessarily bad, but could be made cleaner.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
- [mime-types](https://www.npmjs.com/package/mime-types?utm_source=chatgpt.com) is the library we use for [mime types.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/MIME_types/Common_types)
- [node.js streams](https://betterstack.com/community/guides/scaling-nodejs/nodejs-streams/) give us a more efficient way of reading files from supabase and writing them to disk.
- [dynamic segments](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes) let us keep one route for all files and change that route dynamically based on file ID.

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
https://github.com/calblueprint/rose-academies-uganda/pull/16

CC: @nehaahussain

### CODE REVIEW

File: app/api/sync/route.ts [hunk 1]
```diff
@@ -1,9 +1,24 @@
 import { NextResponse } from "next/server";
+import fs from "fs";
+import path from "path";
 import Database from "better-sqlite3";
 import supabase from "@/api/supabase/client";
 
+const BUCKET = "lesson-files";
+const LOCAL_DIR = "/home/nathantam/rose-files";
```
Comment (nehaahussain) [inline app/api/sync/route.ts]: this is hardcoded since it directly references the file system path on your computer. 

instead of hardcoding the directory, can you use a dynamic path that adjusts to anyone's environment?

**Review by nehaahussain — CHANGES_REQUESTED** [review]
GREAT JOB! please slack me once you've addressed these comments.

File: app/api/sync/route.ts (hunk 1 above)
Comment (nathandtam) [reply app/api/sync/route.ts]: edited it to be dependent on env with a fallback default to home/pi/rose-files
